File: src/ComVu/MainViewModel.cs
using Reactive.Bindings;
using System.Reactive.Linq;

namespace ComVu
{
    public class MainViewModel
    {
        public ReactiveProperty<string> Code { get; set; }
        public ReactiveProperty<string> Output { get; set; }

        public MainViewModel()
        {
            Code = new ReactiveProperty<string>();
            Output = Code
                .ToReactiveProperty();
        }
    }
}

File: src/ComVu.ViewModels/MainViewModel.fs
namespace ComVu.ViewModels

open System
open Reactive.Bindings
open System.Reactive.Linq
open ComVu

type MainViewModel() =

  let mutable code = new ReactiveProperty<string>()
  let mutable analysis = code.Select(not << String.IsNullOrEmpty).ToReactiveCommand()
  let output = new ReactiveProperty<string>("")

  do
    analysis
      .Subscribe(fun _ ->
        output.Value <-
          match Analyzer.analysis code.Value |> Async.RunSynchronously with
          | Success v ->v.ToString()
          | Failure msgs -> msgs |> String.concat Environment.NewLine
      )
    |> ignore

  member __.Code with get() = code and set(value) = code <- value
  member __.Analysis with get() = analysis and set(value) = analysis <- value
  member __.Output = output
